cdap-data-fabric/src/main/java/io/cdap/cdap/store/StoreDefinition.java

```diff
@@ -82,7 +86,11 @@ private static void createIfNotExists(StructuredTableAdmin admin,
 
     StructuredTableId tableId = spec.getTableId();
     try {
-      admin.createOrUpdate(spec);
+      if (admin.isBatchCreateRequired()) {
+        admin.updateOrDeferCreate(spec);
```

Comment:
updateOrDeferCreate is very confusing. Why not batch both update and create?
and the method can be called deferredUpdateOrCreate() which then makes sense.

Reply:
The issue in batching update calls is that : 
Update logic is bit different than the create. Create just executes the ddl statements and ignores SpannerException saying its a concurrent error.

Whereas in update, it has 2-3 different steps - [here](https://github.com/cdapio/cdap/blob/4f5fbc2eb2e01aa197ddf45c1dbbdc0442115ca1/cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTableAdmin.java#L150C4-L171C80)

It is bit tricky to unify these 2 executions - create and update

Other option I can think of is naming method as handleCreate()
it will call tryUpdating() if table exists, as per the current flow. Else it would persist create sqls in the ddl statements.

Reply:
+1, I'd defer everything. Even if it requires multiple calls done later. Otherwise, it makes interface half-sync, half-async. Also may be instead of introducing a new method, we should do something like a 
```
try (var batch = admin.runBatch()) {
 batch.updateOrCreate();
 batch.execute();
}
```
And implement it for all SPIs by default, so that one don't need to do the "ifs".

Reply:
Currently, all the store definitions follow this path : 
for each store -> Store.create() -> for every table spec -> call createIfNotExists -> admin.CreateOrUpdate

Let me know if the following updates in the above flow would work:

1. Get all the table specs at the start itself. Every store will add a spec to this list:
`List<TableSpec> tables;`

2. Once list is formed, we call `execute(tables);`

3. Execute(List<TableSpec> tables) in case of all implementations except spanner:
```
for (spec : specs) {
  createOrUpdate()
}
```
4. Execute(List<TableSpec> tables) in case of cloud spanner
Note : we only want to batch create. Update should be executed individually and follow the current logic as is.
```
List<String> createTableDdlStatements;
for (spec : specs) {
  if exists -> tryUpdating() // this is individual call
  else -> createTableDdlStatements.add(getCreateDdl()); // Accumulate create statements for execution later.
}
executeAll(createTableDdlStatements);
```

Reply:
Yep, makes sense. Main thing is that interface should not depend on the specific implementation features / limitations. If we want to provide batch interface, it should allow to batch everything / nothing  / or in between depending on specific implementation needs